src/routes/counter/containers/CounterPage.js

```diff
@@ -1,17 +1,21 @@
-const { connect: Connect } = require('react-redux');
+const React = require('react');
+const { useCallback } = require('react');
+const { useSelector } = require('react-redux');
+const { useMiddleEnd } = require('strange-middle-end');
 const CounterPage = require('../components/CounterPage');
-const M = require('../../../middle-end');
 
-const internals = {};
+module.exports = function CounterContainer() {
 
-internals.connect = Connect(
-    (state) => ({
-        counter: M.selectors.counter.getValue(state)
-    }),
-    {
-        increment: () => M.actions.counter.increment({ amount: 1 }),
-        double: () => M.actions.counter.double()
-    }
-);
+    const m = useMiddleEnd();
+    const counter = useSelector(m.selectors.counter.getValue);
+    const handleIncrement = useCallback(() => m.dispatch.counter.increment({ amount: 1 }), [m]);
+    const handleDouble = useCallback(() => m.dispatch.counter.double(), [m]);
```

Comment:
These `useCallback()`s are are only helping you if `CounterPage` is a pure component or otherwise cares about whether the functions that it is passed change over time.  Here's a great article on it: https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
Interesting article! Does this mean we could just leave these as functions, like this, to avoid the unnecessary calls to `React.useCallback(…)`?
```suggestion
    const handleIncrement = () => m.dispatch.counter.increment({ amount: 1 });
    const handleDouble = () => m.dispatch.counter.double();
```

Reply:
I think that is just fine :) `useCallback()` is fantastic when you need it, but I think we can keep it simple here.

Reply:
Added changes here: https://github.com/BigRoomStudios/strangeluv/pull/251/commits/bf7a9ef7fdaf926facf90e2bcd5ff3550359dc10